File: server/dotnet/sdk-dotnet-example/Infrastructure/Mappers/PaymentDetails/AcquirerSelectionInformationMapper.cs
using AcquirerSelectionInformationSdk = OnlinePayments.Sdk.Domain.AcquirerSelectionInformation;
using AcquirerSelectionInformationDto = Business.Domain.Payments.PaymentDetails.AcquirerSelectionInformation;

namespace Infrastructure.Mappers.PaymentDetails;

public static class AcquirerSelectionInformationMapper
{
    public static AcquirerSelectionInformationDto Map(AcquirerSelectionInformationSdk? response)
    {
        return new()
        {
            FallbackLevel = response?.FallbackLevel,
            RuleName = response?.RuleName,
            Result = response?.Result
        };
    }
}

File: server/php/src/Infrastructure/Mappers/PaymentDetails/AcquirerSelectionInformationMapper.php
<?php

namespace OnlinePayments\ExampleApp\Infrastructure\Mappers\PaymentDetails;

use OnlinePayments\ExampleApp\Application\Domain\Payments\PaymentDetails\AcquirerSelectionInformation as AcquirerSelectionInformationDto;
use OnlinePayments\Sdk\Domain\AcquirerSelectionInformation as AcquirerSelectionInformationSdk;

final class AcquirerSelectionInformationMapper
{
    public static function mapFromSdkResponse(?AcquirerSelectionInformationSdk $response): ?AcquirerSelectionInformationDto
    {
        if ($response === null) {
            return null;
        }

        $dto = new AcquirerSelectionInformationDto();
        $dto->fallbackLevel = $response?->fallbackLevel ?? null;
        $dto->ruleName = $response?->ruleName ?? null;
        $dto->result = $response?->result ?? null;

        return $dto;
    }
}
